python/lsst/obs/base/ingest.py

```diff
@@ -318,16 +365,181 @@ def _calculate_dataset_info(self, header, filename):
             "visit_id": False,
         }
 
-        obsInfo = ObservationInfo(header, pedantic=False, filename=filename,
-                                  required={k for k in ingest_subset if ingest_subset[k]},
-                                  subset=set(ingest_subset))
+        if isinstance(header, ObservationInfo):
+            obsInfo = header
+            missing = []
+            # Need to check the required properties are present
+            for property, required in ingest_subset.items():
+                if not required:
+                    continue
+                value = getattr(obsInfo, property)
+                if value is None:
```

Comment:
It's a good thing it raises because that would be telling me there is a typo a few lines above when I list all the properties that ingest requires.  The `None` return value is not me expecting a failure in `getattr` it's me checking that the translation succeeded.

Reply:
Maybe worth a comment to that effect above the getattr?